File: katas/8 kyu/beginner-series-number-2-clock/solution.py
# beginner-series-number-2-clock
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def past(h, m, s):
  # s -> * 1000
  # m -> s * 60
  # h -> s * 3600
  return sum([s*1000 , m*1000*60 , h*1000*3600])
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def past(h, m, s):
  return (3600*h + 60*m + s) * 1000
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(past(0,1,1),61000)
test.assert_equals(past(1,1,1),3661000)
test.assert_equals(past(0,0,0),0)
test.assert_equals(past(1,0,1),3601000)
test.assert_equals(past(1,0,0),3600000)

File: katas/8 kyu/beginner-series-number-2-clock/solution.js
// beginner-series-number-2-clock
/* ------------------------------- MY SOLUTION ------------------------------ */
function past(h, m, s){
  return (h*3600 + m*60 + s) * 1000
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;

const { assert } = require('chai');

describe("Fixed Tests", () => {
  it("Tests", () => {
    assert.strictEqual(past(0,1,1),61000)
    assert.strictEqual(past(1,1,1),3661000)
    assert.strictEqual(past(0,0,0),0)
    assert.strictEqual(past(1,0,1),3601000)
    assert.strictEqual(past(1,0,0),3600000)
  });
});